### PR TITLE
Use IMU measurements to set submodel base state in RDE

### PR DESCRIPTION
**Pull request by isorrentino**
This PR implements what is described in #70 and #95.
In addition, the PR  introduces a code restructure to have the possibility to define more than one measurement dynamics for a given sensor.
The tests have been modified accordingly.

[UPDATE] I added also the possibility of having the friction torques as a measurement for the estimator.

### CODE REVIEW

File: CHANGELOG.md.orig [hunk 1]
```diff
@@ -0,0 +1,522 @@
+# Changelog
```
Comment (GiulioRomualdi) [inline CHANGELOG.md.orig:1]: Can you remove this file?

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ All notable changes to this project are documented in this file.
 - Add the support of QP problems with no constraint in `QPInverseKinematics` and `QPTSID` (https://github.com/ami-iit/bipedal-locomotion-framework/pull/784)
 - Implement `blf-joints-grid-position-tracking` application in `utilities` (https://github.com/ami-iit/bipedal-locomotion-framework/pull/787)
 - Add the possibility to resample the contact in a given contact list (https://github.com/ami-iit/bipedal-locomotion-framework/pull/788)
+- Set submodel states from IMUs in RDE (Use IMU measurements to set submodel base state in RDE)
```
Comment (GiulioRomualdi) [inline CHANGELOG.md:18]: Can you rebase the PR on top of master and move this line to unreleased?

File: CHANGELOG.md.orig (hunk 1 above)
Comment (isorrentino) [reply CHANGELOG.md.orig]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (isorrentino) [reply CHANGELOG.md]: Done

**Comment by isorrentino** [comment]
@GiulioRomualdi after the rebase the CI fails.

**Comment by isorrentino** [comment]
@traversaro I requested also your review to be sure about the implemented formulas. If you can check the code here https://github.com/ami-iit/bipedal-locomotion-framework/blob/b8163a9e1b7f1aad447cce7912501db3d3b3c22a/src/Estimators/src/UkfModel.cpp#L114 that is where I take the imu measurements to set the submodel base states and compute the forward dynamics as discussed in https://github.com/ami-iit/element_sensorless-torque-control/issues/70, it would be great :)

**Comment by traversaro** [comment]
> @traversaro I requested also your review to be sure about the implemented formulas. If you can check the code here
> 
> https://github.com/ami-iit/bipedal-locomotion-framework/blob/b8163a9e1b7f1aad447cce7912501db3d3b3c22a/src/Estimators/src/UkfModel.cpp#L114
> that is where I take the imu measurements to set the submodel base states and compute the forward dynamics as discussed in [ami-iit/element_sensorless-torque-control#70](https://github.com/ami-iit/element_sensorless-torque-control/issues/70), it would be great :)

Th computation seems correct. However, formula in https://github.com/ami-iit/bipedal-locomotion-framework/blob/b8163a9e1b7f1aad447cce7912501db3d3b3c22a/src/Estimators/src/UkfModel.cpp#L154 are not obvious, so if you could add a reference (for example with a comment) for that equation it would be great.

**Review by traversaro — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
Hi @isorrentino can you rebase the PR on top of master?

**Comment by isorrentino** [comment]
> Hi @isorrentino can you rebase the PR on top of master?

Done.

**Review by GiulioRomualdi — APPROVED** [review]

**Pull request merged**